### PR TITLE
Eslint, Material-UI, deployment workflow integration

### PR DESCRIPTION
**Pull request by GalenWong**
* Added eslint
* Fixed styling inconsistency
* Fixed Material-UI styling constraints
* Tested Deployment

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  
+      // react plugin - options
+      "react/jsx-uses-react": "error",
+      "react/jsx-uses-vars": "error",
+    },
+    parser: "babel-eslint",
+    parserOptions: {
+      ecmaVersion: 8, // optional, recommended 6+
+    },
+  }
```
Comment (TimothyGu) [inline .eslintrc.js]: Please reformat this file – add trailing newline, fix indentation, etc.

File: .huskyrc.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    "hooks": {
+        /* you can skip git hooks with --no-verify flag */
+        /* but make sure you know absolutely what you are doing */
+        "pre-commit": "npm run lint",
+        "pre-push": "npm run lint"
+    }
+}
```
Comment (TimothyGu) [inline .huskyrc.js]: Ditto w.r.t. formatting.

File: package.json [hunk 3]
```diff
@@ -29,10 +30,19 @@
     "develop": "gatsby develop",
     "start": "npm run develop",
     "format": "prettier --write \"src/**/*.js\"",
-    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\""
+    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\"",
+    "lint": "./node_modules/.bin/eslint --ext js,jsx ."
   },
   "devDependencies": {
-    "prettier": "^1.15.2"
```
Comment (TimothyGu) [inline package.json]: Why remove prettier? Also even if we do so, fix the "format" script above.

File: public/index.html [hunk 4]
```diff
@@ -1,2 +1 @@
-<!DOCTYPE html><html><head><link href="https://fonts.googleapis.com/css?family=Poppins" rel="stylesheet">
-<meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="ie=edge"/><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/><title data-react-helmet="true"></title><link rel="shortcut icon" href="/icons/icon-48x48.png"/><link rel="manifest" href="/manifest.webmanifest"/><meta name="theme-color" content="#663399"/><script src="/socket.io/socket.io.js"></script></head><body><div id="___gatsby"></div><script src="/commons.js"></script></body></html>
+<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="ie=edge"/><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/><title data-react-helmet="true"></title><link rel="shortcut icon" href="/icons/icon-48x48.png"/><link rel="manifest" href="/manifest.webmanifest"/><meta name="theme-color" content="#663399"/><script src="/socket.io/socket.io.js"></script></head><body><div id="___gatsby"></div><script src="/commons.js"></script></body></html>
```
Comment (TimothyGu) [inline public/index.html]: Please remove this file. It should already be ignored by `.gitignore`.

File: src/pages/playground.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import HomePage from '../components/homepage/homepagetemplate';
 
 export default class PlayGround extends React.Component {
-  render() {
+  render () {
```
Comment (TimothyGu) [inline src/pages/playground.js]: Eww… can we please not do this.

File: src/components/homepage/homepagetemplate.js [hunk 6]
```diff
@@ -24,21 +33,27 @@ const styles = theme => ({
...
+  render () {
     const { classes } = this.props;
     return (
-      <div className={classes.container}>
-        <img className={classes.banner} src={banner}/>
-        <div className={classes.title}>
-          <div className={classes.mainTitle}>Move Fast.</div>
-          <div className={classes.mainTitle}>Build Things.</div>
-          <div className={classes.subTitle}>Start Hacking.</div>
+      <React.Fragment>
```
Comment (TimothyGu) [inline src/components/homepage/homepagetemplate.js]: Reminder: we could just return an array of elements.

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+module.exports = {
+    extends: ["standard"],
```
Comment (TimothyGu) [inline .eslintrc.js]: Why do we use Standard? Can we just use the same style (and config) as MentorQ?

File: .eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      "standard/computed-property-even-spacing": ["error", "even"],
+      "standard/no-callback-literal": ["error", ["cb", "callback"]],
+  
+      // react plugin - options
+      "react/jsx-uses-react": "error",
+      "react/jsx-uses-vars": "error",
+    },
+    parser: "babel-eslint",
+    parserOptions: {
+      ecmaVersion: 8, // optional, recommended 6+
```
Comment (TimothyGu) [inline .eslintrc.js]: Let's do 10 (2019)

File: gatsby-node.js [hunk 9]
```diff
@@ -5,21 +5,20 @@
...
-exports.createPages = async ({ actions: {createPage}, graphql }) => {
+exports.createPages = async ({ actions: { createPage }, graphql }) => {
   const postTemplate = path.resolve('src/components/post/postTemplate.js');
 
   const result = await graphql(`
     {
       allMarkdownRemark(
         sort: { order: DESC, fields: [frontmatter___date] }
-        filter: { fields: {slug: {regex: "\/posts\/[fall|winter|spring]/"}}}
+        filter: { fields: {slug: {regex: "/posts/[fall|winter|spring]/"}}}
```
Comment (TimothyGu) [inline gatsby-node.js]: Unrelated to this PR but… what flavor is this regex? In JavaScript regex what you're looking for is probably `/\/posts\/(?:fall|winter|spring)\//`. Also note, GraphQL requires multiple escapes for backslashes according its documentation, so it should probably look like `regex: "/\\\\/posts\\\\/(?:fall|winter|spring)\\\\//"`

Also, shouldn't we group posts by year also?

File: package.json [hunk 10]
```diff
@@ -29,10 +30,19 @@
     "develop": "gatsby develop",
     "start": "npm run develop",
     "format": "prettier --write \"src/**/*.js\"",
-    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\""
+    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\"",
+    "lint": "./node_modules/.bin/eslint --ext js,jsx ."
```
Comment (TimothyGu) [inline package.json]: No need to do `./node_modules/.bin/`: the path is already added to `PATH`.

File: .eslintrc.js (hunk 7 above)
Comment (GalenWong) [reply .eslintrc.js]: This is the recommended gatsby config file. I can try using the MentorQ one as well.

File: src/components/homepage/homepagetemplate.js (hunk 6 above)
Comment (GalenWong) [reply src/components/homepage/homepagetemplate.js]: if we just return an array here, wouldn't that require us to put a `key` on the elements? does that make sense to put a `key` though?

Comment (TimothyGu) [reply src/components/homepage/homepagetemplate.js]: I don't think `key` is required but let's just stick with fragment then.

**Pull request merged**